### PR TITLE
fix in TOF mc labels (digitization)

### PR DESCRIPTION
**Pull request by noferini**

### CODE REVIEW

File: Detectors/GlobalTracking/src/MatchTOF.cxx [hunk 1]
```diff
@@ -898,7 +898,10 @@ void MatchTOF::selectBestMatches()
       }
       if (!labelOk) {
         // we have not found the track label among those associated to the TOF cluster --> fake match! We will associate the label of the main channel, but negative
-        mOutTOFLabels.emplace_back(labelsTOF[0].getTrackID(), labelsTOF[0].getEventID(), labelsTOF[0].getSourceID(), true);
+	if(! labelsTOF.size())
+	  LOG(ERROR) << "TOF label not found since size of label is zero. This should not happen!!!!";
```
Comment (shahor02) [inline Detectors/GlobalTracking/src/MatchTOF.cxx:902]: Is'not it better to throw an exception here? Logging an error may go unnoticed.

Comment (noferini) [reply Detectors/GlobalTracking/src/MatchTOF.cxx:902]: Do you mean a FATAL?

Comment (noferini) [reply Detectors/GlobalTracking/src/MatchTOF.cxx:902]: sorry.
I guess you mean
 assert(labelsTOF.size());
I will add it.

**Comment by shahor02** [comment]
I meant ``throw std::runtime_error("TOF label not found since size of label is zero. This should not happen!!!!")``.
It will have the same behaviour as fatal, unless one day you will decide to process this exception in a different way.

**Comment by noferini** [comment]
> I meant throw std::runtime_error("TOF label not found since size of label is zero. This should not happen!!!!").
> It will have the same behaviour as fatal, unless one day you will decide to process this exception in a different way.

I did it. Thanks

**Pull request merged**